spring-kafka-docs/src/main/asciidoc/retrytopic.adoc

```diff
@@ -0,0 +1,449 @@
+
+You can specify the suffixes that will be used by the retry and dlt topics.
+
+====
+[source, java]
+----
+@RetryableTopic(retryTopicSuffix = "-my-retry-suffix", dltTopicSuffix = "-my-dlt-suffix")
+@KafkaListener(topics = "my-annotated-topic")
+public void processMessage(MyPojo message) {
+        // ... message processing
+}
+----
+====
+
+====
+[source, java]
+----
+@Bean
+public RetryTopicConfiguration myRetryTopic(KafkaTemplate<String, MyOtherPojo> template) {
+           return RetryTopicConfiguration
+                   .builder()
+                   .retryTopicSuffix("-my-retry-suffix")
+                   .dltTopicSuffix("-my-dlt-suffix")
+                   .create(template);
+           }
+----
+====
+
+The retry suffix also is itself suffixed with the delay for the topic, for example -1000, except for fixed delay configurations where the suffix is -0, -1, etc, and single topic fixed delay configurations which have no suffix besides the first one.
+
+NOTE: The default suffixes are "-retry" and "-dlt", for retry and dlt topics respectively.
+
+==== Dlt Processing
+
+You can specify the method used to process the Dlt for the topic, as well as the behaviour if that processing fails.
+
+To do that you can use the `@DltHandler` annotation in a method of the class with the `@RetryableTopic` annotation(s).
+Note that the same method will be used for all the `@RetryableTopic` annottated methods within that class.
+
+====
+[source, java]
+----
+@RetryableTopic
+@KafkaListener(topics = "my-annotated-topic")
+public void processMessage(MyPojo message) {
+        // ... message processing
+}
+
+@DltHandler
+public void processMessage(MyPojo message) {
+// ... message processing, persistence, etc
+}
+----
+====
+
+The DLT handler method can also be provided through the RetryTopicConfigurationBuilder.dltHandlerMethod(Class, String) method, providing the class and method name that should handle the DLT topic.
+If a bean instance of the provided class is found in the application context that bean is used for Dlt processing, otherwise an instance is created with it's dependencies properly injected.
+
+====
+[source, java]
+----
+@Bean
+public RetryTopicConfigurer myRetryTopic(KafkaTemplate<Integer, MyPojo> template) {
+return RetryTopicConfigurer
+    .builder()
+    .dltProcessor(MyCustomDltProcessor.class, "processDltMessage")
+    .create(template);
+}
+
+@Component
+public class MyCustomDltProcessor {
+
+    private final MyDependency myDependency;
+
+    public MyCustomDltProcessor(MyDependency myDependency) {
+        this.myDependency = myDependency;
+    }
+
+    public void processDltMessage(MyPojo message) {
+       // ... message processing, persistence, etc
+    }
+}
+----
+====
+
+NOTE: If no DLT handler is provided, the default RetryTopicConfigurer.LoggingDltListenerHandlerMethod is used.
+
+==== Dlt Failure Behaviour
+
+Should the Dlt processing fail, there are two possible behaviours available: `ALWAYS_RETRY` and `ABORT`.
```

Comment:
```suggestion
Should the Dlt processing fail, there are two possible behaviors available: `ALWAYS_RETRY` and `FAIL`.
```
US English for consistency (I am from the UK, originally).
We try to avoid words like abort,

Reply:
Done